chainstate/src/detail/mod.rs

```diff
@@ -557,5 +688,125 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
     }
 }
 
+// FIXME: should I move the tests to a separate file?
 #[cfg(test)]
-mod test;
+mod test {
```

Comment:
I would be in favour of this. Let's keep the `mod.rs` file just with implementation, it's already large enough.

Also, the file seems to be removed from the module tree in favour of an inline module but `test.rs` is not deleted as far as I can see.

Reply:
Damn it, I was sure I was replacing an empty module `mod test {}` with my module.

I reverted this and moved my tests to test.rs